core/src/main/java/jenkins/model/ParameterizedJobMixIn.java

```diff
@@ -215,6 +215,9 @@ public final void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp,
         hudson.model.BuildAuthorizationToken.checkPermission(asJob(), asJob().getAuthToken(), req, rsp);
 
         ParametersDefinitionProperty pp = asJob().getProperty(ParametersDefinitionProperty.class);
+        if (!asJob().isBuildable()) {
+            throw HttpResponses.error(SC_INTERNAL_SERVER_ERROR, new IOException(asJob().getFullName() + " is not buildable!"));
```

Comment:
I'm not entirly convicned that a 500 response code here is correct.
This is not an error in the server - it's an error with the request.

As it is a client error (asking to build a job that is not buildable) it should be a 4xx response.

possibly a [409 Conflict ](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.10)